MastodonSDK/Sources/MastodonUI/View/Container/MediaGridContainerView.swift

```diff
@@ -200,74 +201,96 @@ extension MediaGridContainerView {
-                containerVerticalStackView.addArrangedSubview(bottomHorizontalStackView)
-                bottomHorizontalStackView.addArrangedSubview(mediaViews[2])
-                bottomHorizontalStackView.addArrangedSubview(mediaViews[3])
-            case 5...9:
-                let topHorizontalStackView = createStackView(axis: .horizontal)
-                containerVerticalStackView.addArrangedSubview(topHorizontalStackView)
-                topHorizontalStackView.addArrangedSubview(mediaViews[0])
-                topHorizontalStackView.addArrangedSubview(mediaViews[1])
-                topHorizontalStackView.addArrangedSubview(mediaViews[2])
-                
-                func mediaViewOrPlaceholderView(at index: Int) -> UIView {
-                    return index < mediaViews.count ? mediaViews[index] : UIView()
-                }
-                let middleHorizontalStackView = createStackView(axis: .horizontal)
-                containerVerticalStackView.addArrangedSubview(middleHorizontalStackView)
-                middleHorizontalStackView.addArrangedSubview(mediaViews[3])
-                middleHorizontalStackView.addArrangedSubview(mediaViews[4])
-                middleHorizontalStackView.addArrangedSubview(mediaViewOrPlaceholderView(at: 5))
-                
-                if count > 6 {
-                    let bottomHorizontalStackView = createStackView(axis: .horizontal)
-                    containerVerticalStackView.addArrangedSubview(bottomHorizontalStackView)
-                    bottomHorizontalStackView.addArrangedSubview(mediaViewOrPlaceholderView(at: 6))
-                    bottomHorizontalStackView.addArrangedSubview(mediaViewOrPlaceholderView(at: 7))
-                    bottomHorizontalStackView.addArrangedSubview(mediaViewOrPlaceholderView(at: 8))
-                }
-            default:
-                assertionFailure()
-                return
-            }
+			
+			if count<2 || count>maxCount {
+				assertionFailure("unexpected attachment count \(count)")
+				return
+			}
+			
+			let layoutView = GridLayoutView()
+			layoutView.translatesAutoresizingMaskIntoConstraints = false
+			view.addSubview(layoutView)
+			layoutView.pinToParent()
+			for mediaView in mediaViews {
+				layoutView.addSubview(mediaView)
+			}
+			layoutView.prepare(layout: layout, maxSize: maxSize)
             
             let containerWidth = maxSize.width
-            let containerHeight = count > 6 ? containerWidth : containerWidth * 2 / 3
+			let containerHeight = CGFloat(layoutView.getMeasuredHeight())
             NSLayoutConstraint.activate([
                 view.widthAnchor.constraint(equalToConstant: containerWidth).priority(.required - 1),
                 view.heightAnchor.constraint(equalToConstant: containerHeight).priority(.required - 1),
             ])
         }
     }
 }
+
+class GridLayoutView : UIView {
+	private var layout: MediaLayoutResult?
+	private var measuredHeight = 0
```

Comment:
You can use `private(set)` to make the property readonly to external code without needing a getter:
```suggestion
	private(set) var measuredHeight = 0
```